src/Chapter4.hs

```diff
@@ -489,6 +502,18 @@ Implement the 'Applicative' instance for our 'List' type.
   type.
 -}
 
+instance Applicative List where
+    pure :: a -> List a
+    pure x = Cons x Empty
+
+    (<*>) :: List (a -> b) -> List a -> List b
+    Empty <*> _ = Empty
+    _ <*> Empty = Empty
+    Cons f fs <*> l = (fmap f l) +++ (fs <*> l)
+
+(+++) :: List a -> List a -> List a
+Empty +++ ys = ys
```

Comment:
Perfect name 👌🏼 


You can one more case to make your solution even more efficient:

```haskell
xs +++ Empty = xs
```

In that case, you won't need to walk through each element of the first list when the second one is empty and return the list instantly 🚤